### ISSUE TITLE
Problem with omega index computation

### ISSUE THREAD

**Issue by ksadorf** [aaronmcdaid/Overlapping-NMI#2]
Hi,

First of all, thank you for this implementation. 
I was testing the omega index computation and I faced this assertion error:
``
onmi: onmi.cpp:417: std::pair<double, double> omega(const NodeToGroup&, const NodeToGroup&): Assertion 'numerator >= 0' failed.
``

I am able to compute the lfkNMI with these input file [file1](https://github.com/aaronmcdaid/Overlapping-NMI/files/320774/com_to_nodes_1.txt),[file2](https://github.com/aaronmcdaid/Overlapping-NMI/files/320775/com_to_nodes_2.txt)  but not the omega index.
However, the program works if I passed the same file twice.
Do you have any clues on where the problem could be?

Side remark:
With gcc version 6.1.1 20160501 (GCC), I have some warnings concerning auto_ptr:
 warning: ‘template<class> class std::auto_ptr’ is deprecated
I compiled from another computer with gcc version 4.7.2 (Debian 4.7.2-5). The warnings are not present but the issue is.

Noé

**Comment by aaronmcdaid** [comment]
Hi Noé,
I'm on vacation at the moment, so I won't be able to reply properly until
Tuesday.

If I forget to reply again, doing hesitate to remind me to do so.

Thanks for your email, it's good to know that some people are using my
software!

Have a good weekend
Aaron

**Comment by ksadorf** [comment]
Hi Aaron,

Sorry for the long delay.
PhD defence is somehow time consuming but now I have more time.
Is there anything I can do to help resolving this issue?

Noé

**Comment by aaronmcdaid** [comment]
Hi Noé,
I can now replicate the issue on my machine. I'm not sure exactly what the problem is yet, but I hope to understand it over the weekend.

Those warnings about 'auto_ptr' are frustrating. I will update my code to make it more modern and have no warnings - my knowledge of C++ has improved over the last few years!

**Comment by aaronmcdaid** [comment]
I've pushed some changes that fix the issue.

That assertion can simply be removed,  `assert(numerator<0)`. When I first wrote that code, I hadn't realized that the Omega index can be negative and therefore that this check is incorrect. Your sample data results in Omega = -0.178813

I have also made many other changes to simplify the code and make it a little more modern. I am now using the "range-based for loop" to iterate over containers, instead of my own custom `forEach` macro. This requires that the compiler support `-std=c++11`, but any decent compiler in the last few years supports it. (Anything since g++ 4.8, I think).
